### PR TITLE
[WorkspaceHub] Update enable isolation flag and purge flag support for workspaceHub

### PR DESCRIPTION
**Pull request by ZhidaLiu**
# Description

1. Add data_isolation_enabled support for workspaceHub.
2. Add permenant_delete flag support for workspaceHub delete operation.
3. Add missing changelog for workspaceHub/Project workspace support.

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md).**

## General Guidelines and Best Practices
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-python/blob/main/CONTRIBUTING.md##building-and-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_workspace_hub/workspace_hub.py [hunk 1]
```diff
@@ -44,6 +44,7 @@ def __init__(
         public_network_access: Optional[str] = None,
         identity: Optional[IdentityConfiguration] = None,
         primary_user_assigned_identity: Optional[str] = None,
+        enable_data_isolation: bool = False,
```
Comment (MilesHolland) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_workspace_hub/workspace_hub.py:47]: Please double check if optional bools are supposed to be wrapped in Optional[] like the other fields here. Also, please add a docstring somewhere around line 88 below.

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_workspace_hub/workspace_hub_config.py [hunk 2]
```diff
@@ -20,7 +20,7 @@ def __init__(
         additional_workspace_storage_accounts: Optional[List[str]] = None,
         default_workspace_resource_group: Optional[str] = None,
     ) -> None:
-
```
Comment (MilesHolland) [inline sdk/ml/azure-ai-ml/azure/ai/ml/entities/_workspace_hub/workspace_hub_config.py:23]: Unneeded whitespace change.

File: sdk/ml/azure-ai-ml/azure/ai/ml/operations/_schedule_operations.py [hunk 3]
```diff
@@ -328,7 +328,7 @@ def _resolve_monitor_schedule_arm_id(  # pylint:disable=too-many-branches,too-ma
                     target=ErrorTarget.SCHEDULE,
                     error_category=ErrorCategory.USER_ERROR,
                 )
-
```
Comment (MilesHolland) [inline sdk/ml/azure-ai-ml/azure/ai/ml/operations/_schedule_operations.py:331]: Unneeded whitespace change.

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_workspace_hub/workspace_hub_config.py (hunk 2 above)
Comment (MilesHolland) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_workspace_hub/workspace_hub_config.py:23]: Edit: Unless this was automatically done by a linter. What that this case for this and the _schedule_operations.py file?

File: sdk/ml/azure-ai-ml/azure/ai/ml/operations/_workspace_hub_operation.py [hunk 4]
```diff
@@ -164,7 +164,9 @@ def deserialize_callback(rest_obj):
 
     # @monitor_with_activity(logger, "Hub.BeginDelete", ActivityType.PUBLICAPI)
     @distributed_trace
-    def begin_delete(self, name: str, *, delete_dependent_resources: bool, **kwargs: Dict) -> LROPoller:
+    def begin_delete(
+        self, name: str, *, delete_dependent_resources: bool, permanently_delete: bool = False, **kwargs: Dict
+    ) -> LROPoller:
```
Comment (MilesHolland) [inline sdk/ml/azure-ai-ml/azure/ai/ml/operations/_workspace_hub_operation.py:169]: Add a docstring for permanently_delete here as well please.

File: sdk/ml/azure-ai-ml/azure/ai/ml/_schema/_workspace_hub/workspace_hub.py [hunk 5]
```diff
@@ -40,3 +40,4 @@ class WorkspaceHubSchema(PathAwareSchema):
     managed_network = ExperimentalField(NestedField(ManagedNetworkSchema))
     existing_workspaces = fields.List(fields.Str())
     workspace_hub_config = ExperimentalField(NestedField(WorkspaceHubConfigSchema))
+    enable_data_isolation = fields.Bool()
```
Comment (MilesHolland) [inline sdk/ml/azure-ai-ml/azure/ai/ml/_schema/_workspace_hub/workspace_hub.py:43]: Please modify the appropriate schema unit test to assign and verify this value at least once

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_workspace_hub/workspace_hub.py (hunk 1 above)
Comment (MilesHolland) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_workspace_hub/workspace_hub.py:47]: I just checked - this is correct as it, but it still needs a docstring.

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_workspace_hub/workspace_hub_config.py (hunk 2 above)
Comment (ZhidaLiu) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_workspace_hub/workspace_hub_config.py:23]: reverted

File: sdk/ml/azure-ai-ml/azure/ai/ml/operations/_schedule_operations.py (hunk 3 above)
Comment (ZhidaLiu) [reply sdk/ml/azure-ai-ml/azure/ai/ml/operations/_schedule_operations.py:331]: reverted

File: sdk/ml/azure-ai-ml/azure/ai/ml/operations/_workspace_hub_operation.py (hunk 4 above)
Comment (ZhidaLiu) [reply sdk/ml/azure-ai-ml/azure/ai/ml/operations/_workspace_hub_operation.py:169]: done

File: sdk/ml/azure-ai-ml/azure/ai/ml/_schema/_workspace_hub/workspace_hub.py (hunk 5 above)
Comment (ZhidaLiu) [reply sdk/ml/azure-ai-ml/azure/ai/ml/_schema/_workspace_hub/workspace_hub.py:43]: Updated

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_workspace_hub/workspace_hub_config.py (hunk 2 above)
Comment (ZhidaLiu) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_workspace_hub/workspace_hub_config.py:23]: it's done by running black before, but after revert and run again this doesn't happened, so keep it not changed

**Review by needuv — APPROVED** [review]

File: sdk/ml/azure-ai-ml/azure/ai/ml/operations/_workspace_hub_operation.py [hunk 6]
```diff
@@ -173,6 +175,9 @@ def begin_delete(self, name: str, *, delete_dependent_resources: bool, **kwargs:
             i.e., container registry, storage account, key vault.
             The default is False. Set to True to delete these resources.
         :type delete_dependent_resources: bool
+        :keyword permanently_delete: Workspaces are soft-deleted state by default to allow recovery of workspace data.
```
Comment (MilesHolland) [inline sdk/ml/azure-ai-ml/azure/ai/ml/operations/_workspace_hub_operation.py:178]: Nit keyword -> param
```suggestion
        :parampermanently_delete: Workspaces are soft-deleted state by default to allow recovery of workspace data.
```

**Review by MilesHolland — APPROVED** [review]

File: sdk/ml/azure-ai-ml/azure/ai/ml/operations/_workspace_hub_operation.py [hunk 7]
```diff
@@ -173,6 +175,9 @@ def begin_delete(self, name: str, *, delete_dependent_resources: bool, **kwargs:
             i.e., container registry, storage account, key vault.
             The default is False. Set to True to delete these resources.
         :type delete_dependent_resources: bool
+        :param permanently_delete: Workspaces are soft-deleted state by default to allow recovery of workspace data.
```
Comment (seanyao1) [inline sdk/ml/azure-ai-ml/azure/ai/ml/operations/_workspace_hub_operation.py:178]: remove the word "state" here.

File: sdk/ml/azure-ai-ml/azure/ai/ml/operations/_workspace_hub_operation.py [hunk 8]
```diff
@@ -173,6 +175,9 @@ def begin_delete(self, name: str, *, delete_dependent_resources: bool, **kwargs:
...
+            Set this flag to override the soft-delete behavior and permanently delete your workspace.
```
Comment (seanyao1) [inline sdk/ml/azure-ai-ml/azure/ai/ml/operations/_workspace_hub_operation.py:179]: Set this flag to true to override ...

**Review by seanyao1 — CHANGES_REQUESTED** [review]

**Review by seanyao1 — APPROVED** [review]

**Pull request merged**